src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryDate.cs

```diff
@@ -145,49 +158,59 @@ public static FormulaValue DateDiff(EvalVisitor runner, EvalVisitorContext conte
                     return CommonErrors.RuntimeTypeMismatch(irContext);
             }
 
-            var units = (StringValue)args[2];
+            var timeUnit = ((StringValue)args[2]).Value.ToLowerInvariant();
+            var isSubdayUnit = IsSubdayTimeUnit(timeUnit);
 
             // When converting to months, quarters or years, we don't use the time difference
             // and applying changes to map time to UTC could lead to weird results depending on the local time zone
             // as we could even change of year if the date we have is 1st of Jan and we are in a UTC+N time zone (N positive)
-            switch (units.Value.ToLower())
+            switch (timeUnit)
             {
                 case "months":
                     double months = ((end.Year - start.Year) * 12) + end.Month - start.Month;
                     return new NumberValue(irContext, months);
                 case "quarters":
-                    var quarters = ((end.Year - start.Year) * 4) + Math.Floor(end.Month / 3.0) - Math.Floor(start.Month / 3.0);
+                    var quarters = ((end.Year - start.Year) * 4) + Math.Floor((end.Month - 1) / 3.0) - Math.Floor((start.Month - 1) / 3.0);
```

Comment:
Could you please add a comment here explaining the decrement to start and end months?

Reply:
Added